### ISSUE TITLE
MySQL migration: renamed issues won't open

### ISSUE THREAD

**Issue by imeszaros** [gitbucket/gitbucket#2209]
We just migrated our GB instance (v 4.27) to MySQL.

At first we wanted to use MySQL 8.x, but reverted to 5.7.24 because of #2150.

GB DB export successfully resulted in a script more than 60k lines, so i needed to use the 'mysql' tool to import it to the new DB.

After that, GB was up and running properly, but opening some issues caused 500 with the following trace:

```
13:32:02.261 [qtp12087621-58] ERROR g.core.controller.IssuesController - Catch unhandled error in request: Request(GET //maltea0095:1099/devs/status-report-frontend/issues/48)@435895
java.lang.ArrayIndexOutOfBoundsException: 1
   at gitbucket.core.issues.html.commentlist$.$anonfun$apply$7(commentlist.template.scala:247)
   at scala.collection.immutable.List.map(List.scala:283)
   at gitbucket.core.issues.html.commentlist$.apply(commentlist.template.scala:93)
   at gitbucket.core.issues.html.issue$.apply(issue.template.scala:72)
   at gitbucket.core.controller.IssuesControllerBase.$anonfun$$init$$7(IssuesController.scala:102)
...
```

After some analysis i found out that the problem only occurred with issues renamed in the past. The exact problem is that during the migration process the line delimiter between the old and the new issue names changed from `\r\n` to `\n`.

With these updates i could solve the issue:

```
update issue_comment set CONTENT = replace(CONTENT, '\n', '\r\n') where `ACTION` = 'change_title';
update issue_comment set CONTENT = replace(CONTENT, '\r\r', '\r') where `ACTION` = 'change_title';
```

We're using `utf8 unicode_ci` in MySQL.

**Comment by takezoe** [comment]
Thanks for your reporting. It should be fixed in GitBucket side to not depend on the line separator.

**Comment by takezoe** [comment]
#2111 will fix `ArrayIndexOutOfBoundsException`.

**Comment by PERAGIT** [comment]
Hello, we are trying to use gitbucket for the first time and we are seeing the same issue:
Failed to add the foreign key constraint. Missing index for constraint 'IDX_ISSUE_COMMENT_FK0' in the referenced table 'issue' when using mysql 8.0

Using the latest version of gitbucket available 4.29.  Any ideas how to make it work with this version?

Thank you.
